python-api/gstswitch/controller.py

```diff
@@ -394,3 +424,57 @@ def parse_preview_ports(cls, res):
         for tupl in liststr:
             preview_ports.append(int(tupl[0]))
         return preview_ports
+
+    def on_preview_port_added(self, callback):
+        """Register a Callback for the preview_port_added Signal
```

Comment:
These docstrings should tell you what "preview_port_added" actually means.
They should also include documentation on the argument signature of the callback.